node/out_response.js

```diff
@@ -337,6 +352,12 @@ TChannelOutResponse.prototype.send = function send(res1, res2) {
         ));
     }
 
+    // TODO: may be spam, consider dropping
+    if (self.inreq.connection.closing) {
+        self.logger.info('ignoring outresponse.send on a closed connection', self.extendLogInfo({}));
+        return;
+    }
+
     self.sendCallResponseFrame([self.arg1, res1, res2], true);
     self.emitFinish();
```

Comment:
These early returns above preclude this emitFinish(). Do all of these cases imply that emitFinish() has already been sent? Perhaps we should emitFinish() up top if it hasn’t been emitted yet.

Also, callers of sendError may be expecting it to return self consistently.